internal/command/resource_ower_model.go

```diff
@@ -0,0 +1,44 @@
+package command
+
+import (
+	"context"
+
+	"github.com/zitadel/zitadel/internal/api/authz"
+	"github.com/zitadel/zitadel/internal/eventstore"
+)
+
+// resourceOwnerModel can be used to retrieve the resourceOwner of an aggregate
+// by checking the first event it.
+type resourceOwnerModel struct {
+	instanceID    string
+	aggregateType eventstore.AggregateType
+	aggregateID   string
+
+	resourceOwner string
+}
+
+func NewResourceOwnerModel(ctx context.Context, aggregateType eventstore.AggregateType, aggregateID string) *resourceOwnerModel {
+	return &resourceOwnerModel{
+		instanceID:    authz.GetInstance(ctx).InstanceID(),
```

Comment:
can we move the context read outside of the model? just remove the context from the parameters